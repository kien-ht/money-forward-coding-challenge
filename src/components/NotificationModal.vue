<template>
  <transition name="modal">
    <b-modal v-model="show" :title="content.title" centered>
      <p class="mb-2">
        {{ content.body }}
      </p>
      <template v-slot:modal-footer>
        <b-button variant="secondary" @click="close">Close</b-button>
      </template>
    </b-modal>
  </transition>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "NotificationModal",

  data() {
    return {
      show: false,
      content: {
        title: "",
        body: "",
      },
    };
  },

  methods: {
    open(options: { title: string; body: string }) {
      if (options) this.content = options;
      this.show = true;
    },
    close() {
      this.show = false;
    },
  },
});
</script>

<style lang="scss" scoped></style>
