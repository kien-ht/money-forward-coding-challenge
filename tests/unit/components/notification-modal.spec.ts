import { shallowMount, mount } from "@vue/test-utils";
import NotificationModal from "@/components/NotificationModal.vue";

describe("NotificationModal.vue", () => {
  it("snapshot", () => {
    const wrapper = mount(NotificationModal, {});
    expect(wrapper).toMatchSnapshot();
  });

  it("renders notification content", () => {
    const content = {
      title: "test title",
      body: "test body",
    };
    const wrapper = shallowMount(NotificationModal, {
      data: () => ({ content }),
    });
    const modalTitle = wrapper.findComponent({ name: "b-modal" }).props().title;
    const modalBody = wrapper.find("p").text();
    expect(modalTitle).toBe(content.title);
    expect(modalBody).toBe(content.body);
  });
});
