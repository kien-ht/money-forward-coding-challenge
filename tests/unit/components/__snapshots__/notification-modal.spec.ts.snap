// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationModal.vue snapshot 1`] = `
<transition-stub name="modal">
  <!---->
</transition-stub>
`;
